packages/components/src/internal/components/chart/utils.ts

```diff
@@ -90,31 +101,57 @@ export function getBarChartPlotConfig(props: BarChartPlotConfigProps): Record<st
 
         scales['color'] = {
             scaleType: 'discrete',
+            sortFn: vis.discreteSortFn,
             scale: function (key) {
                 return barFillColors[key] || defaultFillColor;
             },
         };
     }
 
+    if (grouped) {
+        aes['x'] = 'xSub';
+        aes['xSub'] = 'x';
+        aes['color'] = 'x';
```

Comment:
Do I remember correctly that `aes` stands for `aesthetics`?  I always feel like I need a small decoder ring when I look at this vis stuff, but that would still be true if this were called `aesthetics`. Why does `color` equal `x`?  I believe this is saying to get the color from the 'x' dimension, right?  Maybe comments here would be worthwhile, but maybe if I knew more about what's on the other end I would have less need to accept this on faith.

Reply:
yes, I believe it stands for aesthetics. I added a comment about the x / xSub / color variable usages and what they mean for the stacked bar case